### PR TITLE
Report cookies task

### PR DESCRIPTION
**Pull request by karthikuj**
- Reports cookies on change
- Now reports correct URL, using the tabs API.

Signed-off-by: Karthik UJ [karthikuj2001@gmail.com](mailto:karthikuj2001@gmail.com)

### CODE REVIEW

**Comment by psiinon** [comment]
So .. bit cheeky .. but can you add a unit test for this? 
We now have them 😁

**Comment by karthikuj** [comment]
> So .. bit cheeky .. but can you add a unit test for this? 
> We now have them 😁

Sure 🤝😄

File: source/manifest.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
   "permissions": [
     "activeTab",
+    "cookies",
```
Comment (aryangupta701) [inline source/manifest.json:19]: I have one question regarding this. We have only mentioned permission for `activeTab` while in the content script we are accessing all the tabs. Will it work fine ? As `activeTab` only give permission to access the active tab.  
Maybe adding `tabs` in permissions will be good ? 

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#activetab_permission

**Review by aryangupta701 — COMMENTED** [review]
utils.ts and contants.ts have been replicated twice in background tests as well as in content script tests. Since we are not writing any integration tests for background script constants.ts and utils.ts is not required IMO .

**Comment by karthikuj** [comment]
> utils.ts and contants.ts have been replicated twice in background tests as well as in content script tests. Since we are not writing any integration tests for background script constants.ts and utils.ts is not required IMO .

Removed :+1:

**Comment by karthikuj** [comment]
Will fix the linting errors.

**Comment by karthikuj** [comment]
Ping @aryangupta701 @psiinon :)

**Comment by aryangupta701** [comment]
LGTM :)

**Comment by psiinon** [comment]
Presumable a java change will be needed in order for the cookies to be shown in the ZAP Client Map / Details screens?

**Comment by thc202** [comment]
No, that would have to be done in the zap-extensions repo.

Thank you!

**Pull request merged**